expr/shaper.go

```diff
@@ -336,7 +343,16 @@ func (s *Shaper) createShapeSpec(inType, spec *zng.TypeRecord) (shapeSpec, error
 		}
 	}
 	step, err := createStepRecord(inType, typ)
-	return shapeSpec{typ, step}, err
+	var final zng.Type
+	if typ.ID() == s.typ.ID() {
+		// If the underlying records are the same, then use the
+		// spec record as it might be an alias and the intention
+		// would be to cast to the named type.
+		final = s.typ
+	} else {
+		final = typ
+	}
+	return shapeSpec{final, step}, err
```

Comment:
```suggestion
	final := zng.Type(typ)
	if typ.ID() == s.typ.ID() {
		// If the underlying records are the same, then use the
		// spec type as it might be an alias and the intention
		// would be to cast to the named type.
		final = s.typ
	}
	return shapeSpec{final, step}, err
```